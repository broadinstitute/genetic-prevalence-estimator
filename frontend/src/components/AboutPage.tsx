import {
  Box,
  Breadcrumb,
  BreadcrumbItem,
  BreadcrumbLink,
  Heading,
  Link,
  Text,
} from "@chakra-ui/react";
import { Link as RRLink } from "react-router-dom";

import DocumentTitle from "./DocumentTitle";

const AboutView = () => {
  return (
    <>
      <Text mb={2}>
        The genetic prevalence tool uses variant databases and population data
        to estimate prevalence for recessive diseases. This tool uses data from
        the open source databases{" "}
        <Link href="https://www.ncbi.nlm.nih.gov/clinvar/" isExternal>
          ClinVar
        </Link>{" "}
        and{" "}
        <Link href="https://gnomad.broadinstitute.org/" isExternal>
          gnomAD
        </Link>
        .
      </Text>

      <Text>
        The source code necessary to reproduce the analyses, is available on{" "}
        <Link
          href="https://github.com/broadinstitute/aggregate-frequency-calculator"
          isExternal
        >
          GitHub
        </Link>
        .
      </Text>

      <Heading as="h2" size="md" mt={4} mb={2}>
        Funding
      </Heading>
      <Text>
        Financial support for this tool is provided by the{" "}
        <Link href="https://the-tgg.org/" isExternal>
          Broad Institute
        </Link>{" "}
        and the{" "}
        <Link
          href="https://chanzuckerberg.com/science/programs-resources/rare-as-one/"
          isExternal
        >
          Chan Zuckerberg Initiative
        </Link>
        .
      </Text>

      <Heading as="h2" size="md" mt={4} mb={2}>
        Data Use
      </Heading>
      <Text>
        All data here are released openly and publicly for the benefit of the
        wider biomedical community. There are absolutely no restrictions or
        embargoes on the publication of results derived from this tool. The data
        pulled from gnomAD and this tool are available free of restrictions
        under the{" "}
        <Link
          href="https://creativecommons.org/publicdomain/zero/1.0/"
          isExternal
        >
          Creative Commons Zero Public Domain Dedication
        </Link>
        . This means that you can use it for any purpose without legally having
        to give attribution. However, we request that you actively acknowledge
        and give attribution to the gnomAD project
      </Text>

      <Heading as="h2" size="md" mt={4} mb={2}>
        Citation in Publications
      </Heading>
      <Text mb={2}>
        We request that any use of data obtained from the aggregate frequency
        calculator cite the online resource and include a link to the browser.
      </Text>
      <Text>
        There is no need to include us as authors on your manuscript, unless we
        contributed specific advice or analysis for your work.
      </Text>

      <Heading as="h2" size="md" mt={4} mb={2}>
        Contact
      </Heading>
      <Text>
        Please report any issues using the{" "}
        <Link
          href="https://github.com/broadinstitute/aggregate-frequency-calculator/issues"
          isExternal
        >
          issue tracker
        </Link>
        . You can also reach us at{" "}
        <Link href="mailto:freq-calc@broadinstitute.org" isExternal>
          freq-calc@broadinstitute.org
        </Link>
        .
      </Text>
    </>
  );
};

const AboutPage = () => {
  return (
    <>
      <DocumentTitle title="About" />

      <Box mb={2}>
        <Breadcrumb>
          <BreadcrumbItem>
            <BreadcrumbLink as={RRLink} to="/">
              Home
            </BreadcrumbLink>
          </BreadcrumbItem>
          <BreadcrumbItem isCurrentPage>
            <span>About</span>
          </BreadcrumbItem>
        </Breadcrumb>
      </Box>
      <Heading as="h1" mb={4}>
        About
      </Heading>

      <AboutView />
    </>
  );
};

export default AboutPage;
