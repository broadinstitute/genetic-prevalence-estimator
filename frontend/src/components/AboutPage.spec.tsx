import { expect, test } from "@jest/globals";
import renderer from "react-test-renderer";
import { withDummyRouter } from "../../tests/__helpers__/router";

import AboutPage from "./AboutPage";

test("About Page has no unexpected changes", () => {
  const tree = renderer.create(withDummyRouter(<AboutPage />));
  expect(tree).toMatchSnapshot();
});
