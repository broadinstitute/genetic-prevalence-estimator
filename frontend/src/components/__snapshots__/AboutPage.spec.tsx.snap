// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Page has no unexpected changes 1`] = `
Array [
  <div
    className="css-wlea3r"
  >
    <nav
      aria-label="breadcrumb"
      className="chakra-breadcrumb css-0"
    >
      <ol
        className="chakra-breadcrumb__list css-0"
      >
        <li
          className="chakra-breadcrumb__list-item css-18biwo"
        >
          <a
            className="chakra-breadcrumb__link css-0"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
          <span
            className="css-t4q1nq"
            role="presentation"
          >
            /
          </span>
        </li>
        <li
          className="chakra-breadcrumb__list-item css-18biwo"
        >
          <span>
            About
          </span>
        </li>
      </ol>
    </nav>
  </div>,
  <h1
    className="chakra-heading css-1itv5e3"
  >
    About
  </h1>,
  <p
    className="chakra-text css-wlea3r"
  >
    GeniE, the Genetic Prevalence Estimator, uses variant databases and population data to estimate prevalence for recessive diseases. This tool uses data from the open source databases

    <a
      className="chakra-link css-0"
      href="https://www.ncbi.nlm.nih.gov/clinvar/"
      rel="noopener noreferrer"
      target="_blank"
    >
      ClinVar
    </a>

    and

    <a
      className="chakra-link css-0"
      href="https://gnomad.broadinstitute.org/"
      rel="noopener noreferrer"
      target="_blank"
    >
      gnomAD
    </a>
    .
  </p>,
  <p
    className="chakra-text css-0"
  >
    The source code necessary to reproduce the analyses is available on

    <a
      className="chakra-link css-0"
      href="https://github.com/broadinstitute/genetic-prevalence-estimator"
      rel="noopener noreferrer"
      target="_blank"
    >
      GitHub
    </a>
    .
  </p>,
  <h2
    className="chakra-heading css-1oiljpd"
  >
    Funding
  </h2>,
  <p
    className="chakra-text css-0"
  >
    Financial support for this tool is provided by the

    <a
      className="chakra-link css-0"
      href="https://the-tgg.org/"
      rel="noopener noreferrer"
      target="_blank"
    >
      Broad Institute
    </a>

    and the

    <a
      className="chakra-link css-0"
      href="https://chanzuckerberg.com/science/programs-resources/rare-as-one/"
      rel="noopener noreferrer"
      target="_blank"
    >
      Chan Zuckerberg Initiative
    </a>
    .
  </p>,
  <h2
    className="chakra-heading css-1oiljpd"
  >
    Data Use
  </h2>,
  <p
    className="chakra-text css-0"
  >
    All data here are released openly and publicly for the benefit of the wider biomedical community. There are absolutely no restrictions or embargoes on the publication of results derived from this tool. The data pulled from gnomAD and this tool are available free of restrictions under the

    <a
      className="chakra-link css-0"
      href="https://creativecommons.org/publicdomain/zero/1.0/"
      rel="noopener noreferrer"
      target="_blank"
    >
      Creative Commons Zero Public Domain Dedication
    </a>
    . This means that you can use it for any purpose without legally having to give attribution. However, we request that you actively acknowledge and give attribution to the gnomAD project
  </p>,
  <h2
    className="chakra-heading css-1oiljpd"
  >
    Citation in Publications
  </h2>,
  <p
    className="chakra-text css-wlea3r"
  >
    We request that any use of data obtained from GeniE cite the online resource and include a link to the tool (
    localhost
    ).
  </p>,
  <p
    className="chakra-text css-0"
  >
    There is no need to include us as authors on your manuscript, unless we contributed specific advice or analysis for your work.
  </p>,
  <h2
    className="chakra-heading css-1oiljpd"
  >
    Contact
  </h2>,
  <p
    className="chakra-text css-0"
  >
    Please report any issues using the

    <a
      className="chakra-link css-0"
      href="https://github.com/broadinstitute/genetic-prevalence-estimator/issues"
      rel="noopener noreferrer"
      target="_blank"
    >
      issue tracker
    </a>
    . You can also reach us at

    <a
      className="chakra-link css-0"
      href="mailto:prev-genie@broadinstitute.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      prev-genie@broadinstitute.org
    </a>
    .
  </p>,
]
`;
